DEVELOPER_GUIDE.md

```diff
@@ -494,3 +497,15 @@ Pass a list of files or directories to limit your search.
 The Github workflow in [lucene-snapshots.yml](.github/workflows/lucene-snapshots.yml) is a Github worfklow executable by maintainers to build a top-down snapshot build of lucene.
 These snapshots are available to test compatibility with upcoming changes to Lucene by updating the version at [version.properties](buildsrc/version.properties) with the `version-snapshot-sha` version.
 Example: `lucene = 10.0.0-snapshot-2e941fc`.
+
+# Creating uber-JAR/fat-JAR of a module
```

Comment:
Make this "Creating Fat JARs", a bit shorter

Let's put this under Gradle Build, so more than 1 ## and move it up?